src/test/ui/consts/min_const_fn/min_const_fn_dyn.rs

```diff
@@ -7,9 +7,9 @@ struct Hide(HasDyn);
 const fn no_inner_dyn_trait(_x: Hide) {}
 const fn no_inner_dyn_trait2(x: Hide) {
     x.0.field;
-//~^ ERROR trait bounds other than `Sized`
+//~^ ERROR trait objects in const fn are unstable
```

Comment:
For the cases of fields like this one it would be nice to have a `span_note` pointing at the type and field's type to show how it is a trait object.

Reply:
It would be really nice! Can we implement this though? It seems like we only check locals and not expressions: https://github.com/rust-lang/rust/blob/f84000d9bc93f7c09243144d1a729448da3d714d/compiler/rustc_const_eval/src/transform/check_consts/check.rs#L245-L259

Reply:
I'm not sure if we can't get that info from `local.local_info` or `local.source_info` (I'm not _that_ familiar with const eval and have to relearn it every time I look at it).

Reply:
Hm, after reading docs I can't see how we can get the expression, but maybe I'm missing something

Reply:
Can you check what the contents of `local.local_info` are? I assume it is going to be a `VarBindingForm`. If it is, what are the contents of `opt_match_place`?

Reply:
So in the case of our interest (field access) `local.local_info` is simply `None` :(

Reply:
In that case, we'll have to make more dramatic changes so that _we can_ track that data :-/

It's probably not high priority, but if you desire to take that on, I'd be more than happy to see it through!

Reply:
I don't think I'm interested enough in that. Thanks for offering help though!